server/models/Request.js

```diff
@@ -0,0 +1,33 @@
+const mongoose = require("mongoose");
+
+const requestSchema = new mongoose.Schema({
+  userId: {
+    type: mongoose.Schema.Types.ObjectId,
+    required: true,
+    ref: "User",
+  },
+  sitterId: {
+    type: mongoose.Schema.Types.ObjectId,
+    required: true,
+    ref: "User",
+  },

```

Comment:
This file is not relevant to this PR. There are a couple of ways we can go about using Git when our features depend on other features in development:

- **For simpler cases**: If the depended updates are small enough (e.g. few files/folders) and aren't going to be mutually modified, we can manually check them out but not stage them (e.g. use `git checkout depended-feature-branch -- path(s)/to/files/folders` but do not `git add` them). These will create local copies of these files and we can develop/test our new feature without a problem.

- In most cases, the solution above won't be feasible (e.g. both branches update the same file(s)). A more robust approach is to branch off your new feature of the depended feature branch (e.g. `git checkout depended-feature && git branch new-feature`). The changes from the dependent feature are going to be present and tracked automatically in this case. 

For the latter approach, use `git rebase depended-branch` often during the development of the new feature to ensure your commit history is always up-to-date and you have all the latest changes from the depended branch. When you create a PR for the new feature, if the depended feature isn't merged in yet, it's advisable to change your base to `depended-branch` instead of `main` to facilitate the review process. Once the depended feature is merged in, we can use `git rebase main` to clean up our commit history and correctly reflect changes exclusive to the new feature (you might have to do a `git push --force-with-lease` in this case). We should then change the base branch back to main for the PR.

Reply:
Thanks @ugur-akin , This is very helpful. I was not aware of branching off the depended feature. I will definitely use this strategy in the future.